### PR TITLE
development.rst: Miscellaneous improvements

### PR DESCRIPTION
**Pull request by lfos**
* Fix a couple of grammar mistakes and improve wording.
* Use proper Git terminology for operations on patches.
* Unify capitalization of "GitHub".
* Add proper formatting to branch names.

### CODE REVIEW

File: docs/development.rst [hunk 1]
```diff
@@ -17,7 +17,7 @@ Contributions
 
 Some guidance for contributors:
 
-- discuss about changes on github issue tracker, IRC or mailing list
+- discuss changes on the GitHub issue tracker, IRC or on the mailing list
```
Comment (ThomasWaldmann) [inline docs/development.rst]: "mailing list" - "on the" is repeated for that, but not for "IRC".

so, guess we do not need to repeat it for "mailing list" either.

File: docs/development.rst [hunk 2]
```diff
@@ -29,19 +29,19 @@ Some guidance for contributors:
...
+      bad commit yet: amend the commit to include your fix or
       merge the fixup commit before pushing.
   - have a nice, clear, typo-free commit comment
   - if you fixed an issue, refer to it in your commit comment
   - follow the style guide (see below)
 
 - if you write new code, please add tests and docs for it
 
-- run the tests, fix anything that comes up
+- run the tests, fix any errors that come up
```
Comment (ThomasWaldmann) [inline docs/development.rst]: no. really anything that comes up was meant.

test errors, test failures, pep8 style issues, stuff that you will notice by having a second look, ...

File: docs/development.rst (hunk 1 above)
Comment (lfos) [inline docs/development.rst]: I would rather also say "on IRC" here... But I don't really feel strongly about it :)

File: docs/development.rst (hunk 2 above)
Comment (lfos) [inline docs/development.rst]: Well, "anything" sounds quite strange here. There are "things" (such as the summary) coming up in any case, even if there are no errors. Maybe "fix any issues" instead? I know that this is just nitpicking. But then again, the whole patch is just minor tweaks...

Comment (ThomasWaldmann) [inline docs/development.rst]: yup, change it to "fix any issues that come up".

File: docs/development.rst (hunk 1 above)
Comment (ThomasWaldmann) [inline docs/development.rst]: ok, so "on the GitHub issuetracker, on IRC or on the mailing list".

**Comment by ThomasWaldmann** [comment]
thanks!

**Pull request merged**